### PR TITLE
feature to check if data type is correct under the dataframe

### PR DESCRIPTION
**Pull request by leucir**

### CODE REVIEW

File: iotfunctions/metadata.py [hunk 1]
```diff
@@ -567,7 +567,57 @@ def set_params(self, **params):
...
+    Check if dataframe columns type is equivalent to the data item that is defined in the metadata
+    It checks the entire list of data items. Thus, depending where this code is executed, the dataframe might not be completed.
+    An exception is generated if there are not incompatible types of matching items AND and flag throw_error is set to TRUE
+    '''
+    def check_data_items_type(self, df, items, throw_error=False):
+
+        invalid_data_items = []
+
+        dict_types = {
+            'NUMBER' : ['int64','float64'],
```
Comment (42B) [inline iotfunctions/metadata.py]: Missing whitespace after comma?

File: iotfunctions/metadata.py [hunk 2]
```diff
@@ -567,7 +567,57 @@ def set_params(self, **params):
...
+            logger.info('Dataframe types:')
+            logger.info(df.dtypes)
+
+            for item in list(items.data_items):  #transform in list to iterate over it
+                df_column = {}
+                try:
+                    data_item = items.get(item)  #back to the original dict to retrieve item object
+                    df_column = df[data_item['name']]
+                except KeyError:
+                    logger.debug('Data item %s is not part of the dataframe yet.'% item)
```
Comment (42B) [inline iotfunctions/metadata.py]:
```python
logger.debug('Data item %s is not part of the dataframe yet.', item)  # ?
```

Comment (leucir) [reply iotfunctions/metadata.py]: If the function is invoked in the middle of the pipeline execution, some data items might not have been calculated yet, thus the dataframe won't contain the respective columns. This is only a warning, we can we remove the log if that is the case.

Comment (42B) [reply iotfunctions/metadata.py]: @leucir 

```python
logger.debug('Data item %s is not part of the dataframe yet.'% item)
                                                             ^
logger.debug('Data item %s is not part of the dataframe yet.', item)  # ?
                                                             ^
```

**Comment by leucir** [comment]
@42B @madendorff I added 2 new commits to change the code as suggested by the last review.
Already tested it locally.

**Comment by leucir** [comment]
@42B @madendorff  please, consider the recent commit. This function is now conciliating types, according to the data item (metadata) definition. There is no longer a flag to control the exception, thus if a type conversion is not possible, an exception will be thrown.

**Pull request merged**